src/praisonai-deploy/praisonai_deploy/api.py

```
@@ -149,23 +150,29 @@ def start_api_server(
                 capture_output=True
             )
         except Exception:
             pass
         
+        # Preserve the parent environment (e.g. OPENAI_API_KEY) so the generated
+        # server can reach the agent runtime for /chat.
+        server_env = os.environ.copy()
+        
         # Start server
         if background:
             process = subprocess.Popen(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)